### PR TITLE
Add Support for Atome

### PR DESCRIPTION
**Pull request by jutinapas**
* Add `atome` as a new payment method
* Add support on new fields in Source API for BNPL requests with a related user input form in the Example App

Screenshot_20221223_100940 1 (1) Screenshot_20221222_173554 1 (2) Screenshot_20221223_101216 1 (1)

image

### CODE REVIEW

File: app/src/main/res/xml/preference_payment_setting.xml [hunk 1]
```diff
@@ -227,6 +227,11 @@
             app:key="@string/payment_preference_grabpay_key"
             app:title="@string/payment_preference_grabpay_title" />
 
+        <CheckBoxPreference
+                app:defaultValue="true"
+                app:key="@string/payment_preference_atome_key"
+                app:title="@string/payment_preference_atome_title" />
```
Comment (hpariwat) [inline app/src/main/res/xml/preference_payment_setting.xml:233]: I think the indentation is a bit longer than the above.

File: sdk/src/main/java/co/omise/android/models/Items.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+data class Item(
+    val sku: String? = null,
+    val category: String? = null,
+    val name: String? = null,
+    val quantity: String? = null,
+    val amount: String? = null,
+    val item_uri: String? = null,
+    val image_uri: String? = null,
+    val brand: String? = null,
+) : Parcelable
```
Comment (hpariwat) [inline sdk/src/main/java/co/omise/android/models/Items.kt:16]: EOF

File: sdk/src/main/java/co/omise/android/models/Items.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+@Parcelize
+data class Item(
+    val sku: String? = null,
+    val category: String? = null,
+    val name: String? = null,
+    val quantity: String? = null,
+    val amount: String? = null,
+    val item_uri: String? = null,
+    val image_uri: String? = null,
```
Comment (hpariwat) [inline sdk/src/main/java/co/omise/android/models/Items.kt:14]: Should we use camelCase with jsonProperty like Billing and Shipping?

File: sdk/src/main/res/values-th/strings.xml [hunk 4]
```diff
@@ -107,4 +108,15 @@
     <string name="payment_method_mobile_banking_bay_title">KMA</string>
     <string name="payment_method_mobile_banking_bbl_title">Bualuang mBanking</string>
     <string name="googlepay">Google Pay</string>
-</resources>
\ No newline at end of file
+    <string name="payment_method_atome_title">Atome</string>
+    <string name="label_name">ชื่อ (ไม่จำเป็น)</string>
+    <string name="label_email_optional">อีเมล (ไม่เป็นจำ)</string>
```
Comment (hpariwat) [inline sdk/src/main/res/values-th/strings.xml:113]:
```suggestion
    <string name="label_email_optional">อีเมล (ไม่จำเป็น)</string>
```

Comment (jutinapas) [reply sdk/src/main/res/values-th/strings.xml:113]: Lol, what I have written back there 🤦🏻‍♂️

File: app/src/main/res/xml/preference_payment_setting.xml (hunk 1 above)
Comment (jutinapas) [reply app/src/main/res/xml/preference_payment_setting.xml]: Yeah, all indentations are fixed now, thanks!

File: sdk/src/main/java/co/omise/android/models/Items.kt (hunk 2 above)
Comment (jutinapas) [reply sdk/src/main/java/co/omise/android/models/Items.kt]: Fixed -> https://github.com/omise/omise-android/pull/239/commits/a0220bf3de209bf0ae40408ddcefe8b2bad8e372

File: sdk/src/main/java/co/omise/android/models/Items.kt (hunk 3 above)
Comment (jutinapas) [reply sdk/src/main/java/co/omise/android/models/Items.kt]: Changed -> https://github.com/omise/omise-android/pull/239/commits/a0220bf3de209bf0ae40408ddcefe8b2bad8e372

File: sdk/src/main/java/co/omise/android/ui/AtomeFormFragment.kt [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        ).items(
+            listOf(
+                Item(
+                    "3427842",
+                    "Shoes",
+                    "Prada shoes",
+                    "1",
+                    requester.amount.toString(),
+                    "www.kan.com/product/shoes",
+                    "www.kan.com/product/shoes/image",
```
Comment (kan98) [inline sdk/src/main/java/co/omise/android/ui/AtomeFormFragment.kt:222]: hahahahahaha 😆

File: sdk/src/main/res/values-ja/strings.xml [hunk 6]
```diff
@@ -107,4 +108,15 @@
     <string name="payment_method_mobile_banking_bay_title">KMA</string>
     <string name="payment_method_mobile_banking_bbl_title">Bualuang mBanking</string>
     <string name="googlepay">Google Pay</string>
-</resources>
\ No newline at end of file
+    <string name="payment_method_atome_title">Atome</string>
+    <string name="label_name">Name (optional)</string>
+    <string name="label_email_optional">Email (optional)</string>
+    <string name="label_shipping">Shipping address</string>
+    <string name="label_billing">Billing address</string>
```
Comment (kan98) [inline sdk/src/main/res/values-ja/strings.xml:115]: this is optional

File: sdk/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -123,4 +124,15 @@
     <string name="payment_method_grabpay_title" translatable="false">Grab</string>
     <string name="payment_method_grabpay_rms_title" translatable="false">GrabPay</string>
     <string name="payment_method_grabpay_footnote" translatable="false">(GrabPay and PayLater)</string>
+    <string name="payment_method_atome_title">Atome</string>
+    <string name="label_name">Name (optional)</string>
+    <string name="label_email_optional">Email (optional)</string>
+    <string name="label_shipping">Shipping address</string>
+    <string name="label_billing">Billing address</string>
```
Comment (kan98) [inline sdk/src/main/res/values/strings.xml:131]: optional

File: sdk/src/main/java/co/omise/android/ui/AtomeFormFragment.kt (hunk 5 above)
Comment (jutinapas) [reply sdk/src/main/java/co/omise/android/ui/AtomeFormFragment.kt:222]: Example values ​​you wrote Confluence is something that matches this 😆

File: sdk/src/main/res/values-ja/strings.xml (hunk 6 above)
Comment (jutinapas) [reply sdk/src/main/res/values-ja/strings.xml]: Thanks! -> https://github.com/omise/omise-android/pull/239/commits/7e167db9d49d927cc7fe05f63b6450e7f069d377

File: sdk/src/main/res/values/strings.xml (hunk 7 above)
Comment (jutinapas) [reply sdk/src/main/res/values/strings.xml]: Added -> https://github.com/omise/omise-android/pull/239/commits/7e167db9d49d927cc7fe05f63b6450e7f069d377

File: sdk/src/main/res/values-th/strings.xml [hunk 8]
```diff
@@ -93,6 +93,7 @@
     <string name="truemoney_info_text">กรุณากรอกหมายเลขโทรศัพท์ที่ผูกกับบัญชีทรูมันนี่วอลเล็ท</string>
     <string name="payment_method_fpx_title">FPX</string>
     <string name="fpx_info_text">กรุณากรอกอีเมลเพื่อรับการยืนยันการรับชำระจากเอฟพีเอ็กซ์ (ถ้ามี)</string>
+    <string name="atome_info_text">กรุณากรอกที่จำเป็นเพื่อยืนยันการรับชำระจาก Atome</string>
```
Comment (FhanOmise) [inline sdk/src/main/res/values-th/strings.xml:96]: Maybe?
```suggestion
    <string name="atome_info_text">กรุณากรอกข้อมูลที่จำเป็นเพื่อยืนยันการรับชำระจาก Atome</string>
```

**Review by kan98 — APPROVED** [review]

File: sdk/src/main/res/values-th/strings.xml [hunk 9]
```diff
@@ -107,4 +108,15 @@
...
+    <string name="payment_method_atome_title">Atome</string>
+    <string name="label_name">ชื่อ (ไม่จำเป็น)</string>
+    <string name="label_email_optional">อีเมล (ไม่จำเป็น)</string>
+    <string name="label_shipping">ที่อยู่ในการจัดส่ง</string>
+    <string name="label_billing">ที่อยู่สำหรับส่งใบแจ้งหนี้ (ไม่จำเป็น)</string>
+    <string name="label_street">ที่อยู่</string>
+    <string name="label_postal">รหัสไปรษณีย์</string>
+    <string name="label_city">จังหวัด</string>
+    <string name="label_country_code">รหัสประเทศ</string>
+    <string name="label_billing_shipping">ที่อยู่เดียวกันทั้งจัดส่งและแจ้งหนี้</string>
```
Comment (FhanOmise) [inline sdk/src/main/res/values-th/strings.xml:120]: I think it's a bit informal. May be something like this.
```suggestion
    <string name="label_billing_shipping">ใช้ที่อยู่เดียวกันสำหรับจัดส่งสินค้าและใบแจ้งหนี้</string>
```

**Review by FhanOmise — APPROVED** [review]

File: sdk/src/main/res/values-th/strings.xml (hunk 8 above)
Comment (jutinapas) [reply sdk/src/main/res/values-th/strings.xml:96]: Thanks! 😂

File: sdk/src/main/res/values-th/strings.xml (hunk 9 above)
Comment (jutinapas) [reply sdk/src/main/res/values-th/strings.xml:120]: Sure 😂

**Pull request merged**